packages/components/tree/src/tree.ts

```diff
@@ -0,0 +1,263 @@
+
+  /** Manage keyboard navigation between tabs. */
+  #rovingTabindexController = new RovingTabindexController<TreeNode<T>>(this, {
+    focusInIndex: (elements: Array<TreeNode<T>>) => elements.findIndex(el => !el.disabled),
+    elements: () => Array.from(this.shadowRoot?.querySelectorAll('sl-tree-node') ?? []),
+    isFocusableElement: (el: TreeNode<T>) => !el.disabled
+  });
+
+  /** The virtualizer instance. */
+  #virtualizer?: VirtualizerHostElement[typeof virtualizerRef];
+
+  get dataSource() {
+    return this.#dataSource;
+  }
+
+  /** The model for the tree. */
+  @property({ attribute: false })
+  set dataSource(dataSource: TreeDataSource<T> | undefined) {
+    if (this.#dataSource) {
+      this.#dataSource.removeEventListener('sl-update', this.#onUpdate);
+    }
+
+    this.#dataSource = dataSource;
+    this.#dataSource?.addEventListener('sl-update', this.#onUpdate);
+    this.#dataSource?.update();
+  }
+
+  /** Hides the indentation guides when set. */
+  @property({ type: Boolean, attribute: 'hide-guides' }) hideGuides?: boolean;
+
+  /**
+   * Use this if you want to wait until lit-virtualizer has finished the rendering
+   * the tree nodes. This can be useful in unit tests for example.
+   */
+  get layoutComplete() {
+    return this.#virtualizer?.layoutComplete ?? Promise.resolve();
+  }
+
+  /** Custom renderer function for tree items. */
+  @property({ attribute: false }) renderer?: TreeItemRenderer<T>;
+
+  /**
+   * The custom elements used for rendering this tree. If you are using a custom renderer
+   * to render the tree nodes, any custom elements you use in the renderer need to be specified
+   * via this property. Otherwise those custom elements will not initialize, since the tree
+   * uses a Scoped Custom Element Registry.
+   */
+  @property({ attribute: false }) scopedElements?: Record<string, typeof HTMLElement>;
+
+  /** @internal Emits when the user selects a tree node. */
+  @event({ name: 'sl-select' }) selectEvent!: EventEmitter<SlSelectEvent<TreeDataSourceNode<T>>>;
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    this.role = 'tree';
+  }
+
+  override async firstUpdated(changes: PropertyValues<this>): Promise<void> {
+    super.firstUpdated(changes);
+
+    const wrapper = this.renderRoot.querySelector('[part="wrapper"]') as VirtualizerHostElement;
+    this.#virtualizer = wrapper[virtualizerRef];
+
+    await this.layoutComplete;
+
+    if (this.dataSource?.selection.size) {
+      const node = this.dataSource.selection.keys().next().value as TreeDataSourceNode<T>;
+
+      this.scrollToNode(node, { block: 'center' });
+    }
+  }
+
+  override willUpdate(changes: PropertyValues<this>): void {
+    super.willUpdate(changes);
+
+    if (changes.has('dataSource')) {
+      if (this.dataSource?.selects === 'multiple') {
+        this.setAttribute('aria-multiselectable', 'true');
+      } else {
+        this.removeAttribute('aria-multiselectable');
```

Comment:
Maybe it should be `this.setAttribute('aria-multiselectable', 'false');`? So `aria-multiselectable="false"` by default?

Reply:
Good catch.